### DISCUSSION TITLE
Extract all properties

### DISCUSSION THREAD

**Discussion by plop44** [snoopwpf/snoopwpf#428] [Q&A] [answered]
Hi there,

First of all thanks for your incredible work. I have been using Snoop for years, it's doing a great job!

I wonder if there is currently a way to copy all the grid  values that are highlighted in red in this capture?
![image](https://github.com/snoopwpf/snoopwpf/assets/6773424/072af2d1-6ce6-4f91-b3eb-933319501d35)

I tried CTL+A => CTL+C  with no luck. (it copies one value only)
And nothing either in the context menu that suits.

I think this feature would be really useful as when I am trying to understand how work a control I would like to do:
1/ copy all values
2/ do an action on the control that will change its state
3/ copy all values
4/ drop values from 1/ and 3/ in my favorite Diff tool (KDiff/Winmerge/beyond compare, etc) 

This way I am able to have a full understanding of what changed after the action I did in 2/
I am aware of the yellow highlighting of properties that changed, but I find it not that good when the controls has many properties.

Thanks

**Comment by batzen** [comment, marked as answer]
Have you tried the export function?
I guess that would do exactly what you want.
You can reach it via the context menu of items in the left tree.

**Reply by plop44** [reply]
That is what I was after, thanks!
